worker/main_test.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/pubsub/pstest"
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestGetEnv(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	val := "test"
+	os.Setenv("TEST", val)
+	defaultVal := "default"
+	assert.Equal(val, getEnv("TEST", defaultVal))
+	assert.NotEqual(defaultVal, getEnv("TEST", defaultVal))
+}
+
+func TestGetEnvDefault(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	defaultVal := "default"
+	assert.Equal(defaultVal, getEnv("test", defaultVal))
+}
+
+func TestSubscribe(t *testing.T) {
+	assert := assert.New(t)
+	//create a fake Pub Sub serer
+	ctx := context.Background()
+	// Start a fake server running locally.
+	srv := pstest.NewServer()
+	defer srv.Close()
+	// Connect to the server without using TLS.
+	conn, err := grpc.Dial(srv.Addr, grpc.WithInsecure())
+	assert.Nil(err)
+	defer conn.Close()
+	// Use the connection when creating a pubsub client.
+	client, err := pubsub.NewClient(ctx, "rm-ras-sandbox", option.WithGRPCConn(conn))
+	assert.Nil(err)
+	defer client.Close()
+
+	topic, err := client.CreateTopic(ctx, "test")
+	assert.Nil(err)
+	assert.NotNil(topic)
+	fmt.Println(topic)
+
+	sub, err := client.CreateSubscription(ctx, "sample-workers", pubsub.SubscriptionConfig{
+		Topic: topic,
+	})
+	assert.Nil(err)
+	assert.NotNil(sub)
+	fmt.Println(sub)
+
+	sample := "13110000001:::::::::::WW:::::OFFICE FOR NATIONAL STATISTICS:::::::::0001:"
+
+	s := parse([]byte(sample))
+	sampleJson, err := s.marshall()
+	assert.Nil(err)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Likewise, this might be the sort of thing we want to set up only once if we have future tests that use it https://golang.org/pkg/testing/#hdr-Main

Reply:
Awesome, that's what I wanted. I was trying to find the equivalent of Java's setup and tear down